js/coinw.js

```diff
@@ -0,0 +1,588 @@
+                'contractSize': undefined,
+                'expiry': undefined,
+                'expiryDatetime': undefined,
+                'strike': undefined,
+                'optionType': undefined,
+                'precision': {
+                    'amount': this.parseNumber (this.parsePrecision (amountPrecision)),
+                    'price': this.parseNumber (this.parsePrecision (pricePrecision)),
+                },
+                'limits': {
+                    'leverage': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                    'amount': {
+                        'min': this.safeNumber (market, 'minBuyAmount'),
+                        'max': this.safeNumber (market, 'maxBuyAmount'),
+                    },
+                    'price': {
+                        'min': this.safeNumber (market, 'minBuyPrice'),
+                        'max': this.safeNumber (market, 'maxBuyPrice'),
+                    },
+                    'cost': {
+                        'min': this.safeNumber (market, 'min_order_value'),
+                        'max': undefined,
+                    },
+                },
+                'info': market,
+            });
+        }
+        return result;
+    }
+
+    async fetchCurrencies (params = {}) {
+        /**
+         * @method
+         * @name coinw#fetchCurrencies
+         * @description fetches all available currencies on an exchange
+         * @param {object} params extra parameters specific to the coinw api endpoint
+         * @returns {object} an associative dictionary of currencies
+         */
+        const response = await this.publicGetReturnCurrencies (params);
+        //
+        //  {
+        //      "AGLD":{
+        //         "maxQty":"5714.0",
+        //         "minQty":"3.0",
+        //         "recharge":"1",
+        //         "symbol":"AGLD",
+        //         "symbolId":"569",
+        //         "txFee":"0.0",
+        //         "withDraw":"1"
+        //     },
+        //  }
+        //
+        const data = this.safeValue (response, 'data', {});
+        const dataKeys = Object.keys (data);
+        const dataLength = dataKeys.length;
+        const result = {};
+        for (let i = 0; i < dataLength; i++) {
+            const dataKey = dataKeys[i];
+            const currency = data[dataKey];
+            const symbolId = this.safeString (currency, 'symbolId');
+            const code = this.safeCurrencyCode (this.safeString (currency, 'symbol'));
+            const withdraw = this.safeString (currency, 'withDraw');
+            result[code] = {
+                'id': symbolId,
+                'code': code,
+                'info': currency,
+                'name': undefined,
+                'active': undefined,
+                'deposit': undefined,
+                'withdraw': (withdraw === '1'),
+                'fee': this.safeString (currency, 'txFee'),
```

Comment:
`fee` should not be returned as a string to the user, use `safeNumber` instead pls